packages/frontend/track/src/sentry.ts

```diff
@@ -0,0 +1,50 @@
+import * as Sentry from '@sentry/react';
+import { useEffect } from 'react';
+import {
+  createRoutesFromChildren,
+  matchRoutes,
+  useLocation,
+  useNavigationType,
+} from 'react-router-dom';
+
+function createSentry() {
+  let enabled = true;
+  const wrapped = {
+    init() {
+      // https://docs.sentry.io/platforms/javascript/guides/react/#configure
+      Sentry.init({
+        enabled: enabled,
+        dsn: process.env.SENTRY_DSN,
+        debug: BUILD_CONFIG.debug ?? false,
+        environment: process.env.BUILD_TYPE ?? 'development',
+        integrations: [
+          Sentry.reactRouterV6BrowserTracingIntegration({
+            useEffect,
+            useLocation,
+            useNavigationType,
+            createRoutesFromChildren,
+            matchRoutes,
+          }),
+        ],
+        beforeSend(event) {
+          return enabled ? event : null;
+        },
+      });
+      Sentry.setTags({
+        distribution: BUILD_CONFIG.distribution,
+        appVersion: BUILD_CONFIG.appVersion,
+        editorVersion: BUILD_CONFIG.editorVersion,
+      });
+    },
+    enable() {
+      enabled = true;
+    },
+    disable() {
+      enabled = false;
```

Comment:
Hi @forehalo, just did a bit of testing and unfortunately still seeing Sentry when telemetry is disabled. 

Since the Sentry object has already created, I don't think changing the `enabled` parameter has an effect. Instead, we should probably be updating the Sentry object, like this:

```js
    enable() {
      enabled = true;
      this.init();
    },
    disable() {
      // https://docs.sentry.io/platforms/javascript/guides/react/configuration/draining/
      enabled = false;
      Sentry.getClient()?.close(0)
    },
```

Reply:
I checked sentry source code, it turns out `.close` will make sentry itself unrecoverable from 'disabled' status. will have another try tmr.

Reply:
I took another shot at this in case it's helpful. Here's the relevant [branch](https://github.com/hackeresq/AFFiNE/tree/fix-telemetry-sentry).

Essentially the Sentry BrowserSessions integration doesn't use the `beforeSend()` hook. 
* I created a custom integration that listens for browser session events and blocks them.
* Also added dynamic loading to the integrations only when telemetry is enabled.
* Finally made the default value of `enabled` to be false in the Sentry module until it's set by the AppSettingsHelper.

Interested to hear your thoughts!

Reply:
the `beforeSend` is used but there are `beforeSendSpan` and `beforeSendTransaction` to be overrided but this approach is a little bit verbose. The Integration way is also too complicated. I will do it by just updating options and test pass